core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -0,0 +1,350 @@
+  val tp = new TopicPartition(topic, part)
+  val topicAndPartition = new TopicAndPartition(topic, part)
+  val group = "my-group"
+  val topicResource = new Resource(Topic, topic)
+  val groupResource = new Resource(Group, group)
+
+
+  val authorizer = new SimpleAclAuthorizer
+
+  val consumers = Buffer[KafkaConsumer[Array[Byte], Array[Byte]]]()
+  val producers = Buffer[KafkaProducer[Array[Byte], Array[Byte]]]()
+
+  val numServers = 1
+  val producerCount = 1
+  val consumerCount = 2
+  val producerConfig = new Properties
+  val numRecords = 1
+
+  val overridingProps = new Properties()
+  overridingProps.put(KafkaConfig.AuthorizerClassNameProp, classOf[SimpleAclAuthorizer].getName)
+  overridingProps.put(KafkaConfig.BrokerIdProp, brokerId.toString)
+  overridingProps.put(KafkaConfig.OffsetsTopicPartitionsProp, "1")
+
+  val partitionStateInfo = new PartitionStateInfo(new LeaderIsrAndControllerEpoch(LeaderAndIsr(brokerId,
+    Int.MaxValue, List(brokerId), LeaderAndIsr.initialZKVersion), KafkaController.InitialControllerEpoch), Set(brokerId))
+  val endPoint = new EndPoint("localhost", 0, SecurityProtocol.PLAINTEXT)
+  
+  val RequestKeyToRequest = mutable.LinkedHashMap[Short, RequestOrResponse](
+    RequestKeys.MetadataKey -> new TopicMetadataRequest(List(topic), correlationId),
+    RequestKeys.ProduceKey -> ProducerRequest(ProducerRequest.CurrentVersion, correlationId, clientId, 1, 100,
+      collection.mutable.Map(topicAndPartition -> new ByteBufferMessageSet(ByteBuffer.wrap("test".getBytes)))),
+    RequestKeys.FetchKey -> new FetchRequest(requestInfo = Map(topicAndPartition -> new PartitionFetchInfo(0, 100))),
+    RequestKeys.OffsetsKey -> new OffsetRequest(Map(topicAndPartition -> new PartitionOffsetRequestInfo(-1, 0))),
+    RequestKeys.OffsetCommitKey -> new OffsetCommitRequest(groupId = group, requestInfo = Map(topicAndPartition ->
+      new OffsetAndMetadata(new OffsetMetadata(0)))),
+    RequestKeys.OffsetFetchKey -> new OffsetFetchRequest(groupId = group, requestInfo = Seq(topicAndPartition)),
+    RequestKeys.GroupMetadataKey -> new GroupMetadataRequest(group = group),
+    RequestKeys.UpdateMetadataKey -> UpdateMetadataRequest(UpdateMetadataRequest.CurrentVersion, correlationId, clientId, brokerId,
+      Int.MaxValue, Map(topicAndPartition -> partitionStateInfo), Set(new Broker(brokerId, Map(SecurityProtocol.PLAINTEXT -> endPoint)))),
+    RequestKeys.LeaderAndIsrKey -> new LeaderAndIsrRequest(LeaderAndIsrRequest.CurrentVersion, correlationId, clientId,
+      brokerId, Int.MaxValue, Map((topic, part) -> partitionStateInfo), Set(new BrokerEndPoint(brokerId,"localhost", 0))),
+    RequestKeys.StopReplicaKey -> StopReplicaRequest(StopReplicaRequest.CurrentVersion, correlationId, clientId, brokerId,
+      Int.MaxValue, true, Set(topicAndPartition)),
+    RequestKeys.ControlledShutdownKey -> ControlledShutdownRequest(ControlledShutdownRequest.CurrentVersion, correlationId, Some(clientId), brokerId)
```

Comment:
@Parth-Brahmbhatt , yes, the scala requests/responses will go away in the future. So, it would be better if you could rewrite the test using the java requests/responses. Sorry for not pointing this out earlier.